src/main/java/de/tum/in/www1/artemis/service/notifications/NotificationTargetProvider.java

```diff
@@ -1,53 +1,45 @@
 package de.tum.in.www1.artemis.service.notifications;
 
-import com.google.gson.JsonObject;
+import org.springframework.stereotype.Service;
 
 import de.tum.in.www1.artemis.domain.Course;
 import de.tum.in.www1.artemis.domain.Exercise;
 import de.tum.in.www1.artemis.domain.Lecture;
 import de.tum.in.www1.artemis.domain.ProgrammingExercise;
 import de.tum.in.www1.artemis.domain.metis.Post;
+import de.tum.in.www1.artemis.domain.notification.Notification;
+import de.tum.in.www1.artemis.domain.notification.NotificationTarget;
 
+@Service
```

Comment:
Please do not mix Services with non services. Either this class is a service and is only injected using constructor injection into other services. Then you are not allowed to use the constructor, i.e. no use of

`new NotificationTargetProvider()`

The alternative is, that NotificationTargetProvider is **not** a service, then you use the normal constructor and `new NotificationTargetProvider()` is ok

Reply:
Looking at the methods below, I think NotificationTargetProvider should **not** be a service. It is effectively a factory, so we should call it `NotificationTargetFactory`. All methods should be renamed from `getXTarget` into `createXTarget`

Reply:
I renamed the provider into a factory and adjusted the method names